src/openssl.c

```diff
@@ -4103,6 +4122,31 @@ static int pk_toPEM(lua_State *L) {
 } /* pk_toPEM() */
 
 
+static int pk_getPrivateKey(lua_State *L) {
+	EVP_PKEY *key = checksimple(L, 1, PKEY_CLASS);
+	const char *cname = luaL_optstring(L, 2, NULL);
+	EVP_CIPHER *cipher = NULL;
+	lua_settop(L, 3);
+
+	if (cname) {
+		cipher = EVP_get_cipherbyname(cname);
+		if (!cipher)
+			return luaL_error(L, "pkey:getPrivateKey: unknown cipher: %s", cname);
+	}
+
+	BIO *bio = getbio(L);
+	char *str;
+	long len;
+
+	if (!PEM_write_bio_PrivateKey(bio, key, cipher, NULL, 0, pem_pw_cb, L))
```

Comment:
This doesn't actually encrypt the key: `getbio` pushes a userdata onto the stack which then always results in using no password.